src/nupic/types/Fraction.cpp

```diff
@@ -366,7 +366,7 @@ namespace nupic
 
     while(components.size())
     {
-      std::swap<int>(numerator_, denominator_);
+      std::swap(numerator_, denominator_);
```

Comment:
@scottpurdy It seems that `swap` don't handle `numerator_` and `denominator_` as pointers. According to `cppreference.com`, new templates were added since c++11. Look this example took from it:

```
#include <algorithm>
#include <iostream>
 
int main()
{
   int a = 5, b = 3;
 
   // before
   std::cout << a << ' ' << b << '\n';
 
   std::swap(a,b);
 
   // after
   std::cout << a << ' ' << b << '\n';
}
```

```
Output:

5 3
3 5
```

(http://en.cppreference.com/w/cpp/algorithm/swap)